Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -406,18 +405,72 @@ function New-LDAPSIdentitySource {
     else {
         Write-Host "No existing external identity sources found."
     }
-
     $Password = $Credential.GetNetworkCredential().Password
-    $DestinationFileArray = Get-Certificates -SSLCertificatesSasUrl $SSLCertificatesSasUrl -ErrorAction Stop
-    [System.Array]$Certificates = 
-        foreach($CertFile in $DestinationFileArray) {
+    $DestinationFileArray = @()
+    if ($PSBoundParameters.ContainsKey('SSLCertificatesSasUrl')) {
+        $DestinationFileArray = Get-Certificates -SSLCertificatesSasUrl $SSLCertificatesSasUrl -ErrorAction Stop
+    } else {
+        $exportFolder = "$home/"
+        $remoteComputers = ,$PrimaryUrl
+        if ($PSBoundParameters.ContainsKey('SecondaryUrl')) { $remoteComputers += $SecondaryUrl }
+        
+        foreach ($computerUrl in $remoteComputers) {
+            try {
+                if (![uri]::IsWellFormedUriString($computerUrl, 'Absolute')) { throw }
+                $ParsedUrl = [System.Uri]$computerUrl
+            }
+            catch {
+                throw "Incorrect Url format entered from: $computerUrl"
+            }
+            if ($ParsedUrl.Host -match "^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$" -and [bool]($ParsedUrl.Host -as [ipaddress])) { 
+                throw "Incorrect Url format. $computerUrl is an IP address. Consider changing it to an URL"
+            }
+            
+            $Command = $null
+            $SSHRes = $null
+
             try {
-                [System.Security.Cryptography.X509Certificates.X509Certificate2]::CreateFromCertFile($certfile)
+                $Command = 'nslookup ' + $ParsedUrl.Host + ' -type=soa'
+                $SSHRes = Invoke-SSHCommand -Command $Command -SSHSession $SSH_Sessions['VC']
+                if ($SSHRes.ExitStatus -ne 0) { throw 1 }
+
+                $Command = 'nc -vz ' + $ParsedUrl.Host + ' ' + $ParsedUrl.Port
+                $SSHRes = Invoke-SSHCommand -Command $Command -SSHSession $SSH_Sessions['VC']
+                if ($SSHRes.ExitStatus -ne 0) { throw 2 }
+
+                Write-Host ("Start to Download Cert from " + $computerUrl)
+                $Command = 'echo "1" | openssl s_client -connect ' + $ParsedUrl.Host + ':' + $ParsedUrl.Port + ' -showcerts'
+                $SSHRes = Invoke-SSHCommand -Command $Command -SSHSession $SSH_Sessions['VC']
+                $SSHOutput = $SSHRes.Output | out-string
             } catch {
-                Write-Error "Failure to convert file $certfile to a certificate $($PSItem.Exception.Message)"
-                throw "File to certificate conversion failed. See error message for more details"
+                if ($_.Exception.Message -eq 1) { throw "The FQDN $ParsedUrl.Host cannot be resolved to an IP address" }
+                if ($_.Exception.Message -eq 2) { throw "The port $ParsedUrl.Port is not open" }
```

Comment:
The port is the likely but just one of the possibilities, the issue is a bit more broad - connection cannot be established, which could be due to address, routing and/or firewall. Tailor the message a bit please.